### PR TITLE
Fix default constructor for Orientation, add tests

### PR DESCRIPTION
**Pull request by mar-celine**

### CODE REVIEW

File: src/Domain/Orientation.hpp [hunk 1]
```diff
@@ -24,7 +24,7 @@
 template <size_t VolumeDim>
 class Orientation {
  public:
-  Orientation() = default;
+  Orientation();
```
Comment (nilsdeppe) [inline src/Domain/Orientation.hpp]: add Doxygen comment stating what the default orientation is

File: tests/Unit/Domain/Test_Orientations.cpp [hunk 2]
```diff
@@ -12,12 +12,15 @@ namespace {
...
+  CHECK(default_orientation.is_aligned());
+  CHECK(get_output(default_orientation) == "(+0)");
   Orientation<1> custom1(
       std::array<Direction<1>, 1>{{Direction<1>::upper_xi()}});
-  CHECK(custom1.is_aligned() == true);
+  CHECK(custom1.is_aligned());
   Orientation<1> custom2(
       std::array<Direction<1>, 1>{{Direction<1>::lower_xi()}});
-  CHECK(custom2.is_aligned() == false);
+  CHECK(!custom2.is_aligned());
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_Orientations.cpp]: `CHECK_FALSE`

File: tests/Unit/Domain/Test_Orientations.cpp [hunk 3]
```diff
@@ -39,20 +42,23 @@ void test_1d() {
   // Test semantics:
   const auto custom_copy = custom1;
   test_copy_semantics(custom2);
-  test_move_semantics(std::move(custom1), custom_copy);
+  test_move_semantics(std::move(custom1), custom_copy);  // NOLINT
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_Orientations.cpp]: add comment stating why NOLINT

File: tests/Unit/Domain/Test_Orientations.cpp [hunk 4]
```diff
@@ -39,20 +42,23 @@ void test_1d() {
...
+  CHECK(default_orientation.is_aligned());
+  CHECK(get_output(default_orientation) == "(+0, +1)");
   Orientation<2> custom1(std::array<Direction<2>, 2>{
       {Direction<2>::upper_xi(), Direction<2>::upper_eta()}});
-  CHECK(custom1.is_aligned() == true);
+  CHECK(custom1.is_aligned());
   Orientation<2> custom2(std::array<Direction<2>, 2>{
       {Direction<2>::lower_xi(), Direction<2>::lower_eta()}});
-  CHECK(custom2.is_aligned() == false);
+  CHECK(!custom2.is_aligned());
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_Orientations.cpp]: CHECK_FALSE

File: tests/Unit/Domain/Test_Orientations.cpp [hunk 5]
```diff
@@ -122,10 +128,10 @@ void test_2d() {
         expected_neg_xi_neg_eta_segment_ids);
   CHECK(rotated2d_pos_eta_neg_xi.mapped(segment_ids) ==
         expected_pos_eta_neg_xi_segment_ids);
-  CHECK(rotated2d_neg_eta_pos_xi.is_aligned() == false);
-  CHECK(rotated2d_neg_xi_neg_eta.is_aligned() == false);
-  CHECK(rotated2d_pos_eta_neg_xi.is_aligned() == false);
-  CHECK(rotated2d_pos_xi_pos_eta.is_aligned() == true);
+  CHECK(!rotated2d_neg_eta_pos_xi.is_aligned());
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_Orientations.cpp]: `CHECK_FALSE`s

File: tests/Unit/Domain/Test_Orientations.cpp [hunk 6]
```diff
@@ -143,22 +149,26 @@ void test_2d() {
   // Test semantics:
   const auto rotated_copy = rotated2d_neg_eta_pos_xi;
   test_copy_semantics(rotated2d_pos_eta_neg_xi);
-  test_move_semantics(std::move(rotated2d_neg_eta_pos_xi), rotated_copy);
+  test_move_semantics(std::move(rotated2d_neg_eta_pos_xi),  // NOLINT
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_Orientations.cpp]: why NOLINT?

File: tests/Unit/Domain/Test_Orientations.cpp [hunk 7]
```diff
@@ -143,22 +149,26 @@ void test_2d() {
...
   Orientation<3> custom1(std::array<Direction<3>, 3>{
       {Direction<3>::upper_xi(), Direction<3>::upper_eta(),
        Direction<3>::upper_zeta()}});
-  CHECK(custom1.is_aligned() == true);
+  CHECK(custom1.is_aligned());
   Orientation<3> custom2(std::array<Direction<3>, 3>{
       {Direction<3>::lower_xi(), Direction<3>::lower_eta(),
        Direction<3>::lower_zeta()}});
-  CHECK(custom2.is_aligned() == false);
+  CHECK(!custom2.is_aligned());
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_Orientations.cpp]: CHECK_FALSE

File: tests/Unit/Domain/Test_Orientations.cpp [hunk 8]
```diff
@@ -186,9 +196,9 @@ void test_3d() {
   CHECK(custom_orientation.mapped(2) == 2);
   CHECK(custom_orientation.mapped(direction) == direction.opposite());
   CHECK(custom_orientation.mapped(segment_ids) == flipped_ids);
-  CHECK(custom_orientation.is_aligned() == false);
+  CHECK(!custom_orientation.is_aligned());
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_Orientations.cpp]: CHECK_FALSE

File: tests/Unit/Domain/Test_Orientations.cpp [hunk 9]
```diff
@@ -199,7 +209,7 @@ void test_3d() {
   // Test semantics:
   const auto custom_copy = custom_orientation;
   test_copy_semantics(aligned_orientation);
-  test_move_semantics(std::move(custom_orientation), custom_copy);
+  test_move_semantics(std::move(custom_orientation), custom_copy);  // NOLINT
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_Orientations.cpp]: // clang-tidy: why...?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**